src/webgpu/shader/execution/flow_control/loop.spec.ts

```diff
@@ -0,0 +1,59 @@
+export const description = `
+Flow control tests for loops.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+import { runFlowControlTest } from './harness.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('loop_break')
+  .desc('Test that flow control exits a loop when reaching a break statement')
+  .fn(t => {
+    runFlowControlTest(
+      t,
+      f =>
+        `
+  ${f.expect_sequence(0)}
+  var i = ${f.value(0)};
+  loop {
+    ${f.expect_sequence(1, 3, 5, 7)}
+    if (i == 3) {
+      break;
+    }
+    ${f.expect_sequence(2, 4, 6)}
+    i++;
+  }
+  ${f.expect_sequence(8)}
+`
+    );
+  });
+
+g.test('loop_continue')
+  .desc('Test flow control for a loop continue statement')
+  .fn(t => {
+    runFlowControlTest(
+      t,
+      f =>
+        `
+  ${f.expect_sequence(0)}
+  var i = ${f.value(0)};
+  loop {
+    ${f.expect_sequence(1, 3, 5, 7, 8)}
+    if (i == 3) {
+      i++;
+      continue;
+      ${f.expect_not_reached()}
+    }
+    ${f.expect_sequence(2, 4, 6, 9)}
+    if (i == 4) {
+      break;
+    }
+    i++;
+  }
+  ${f.expect_sequence(10)}
+`
+    );
```

Comment:
Are loop continuing and break-if tests to come later?

Reply:
Yeah, I've started on them, but this isn't meant to be exhaustive at this stage. I wanted to get a framework in place so others can contribute.